### PR TITLE
Fix encryption for IE

### PR DESCRIPTION
**Pull request by gbirke**
IE < 11 doesn't work with the openpgp Javascript encryption library. This PR adds a dynamically loaded, older PGP library from http://www.hanewin.net/encrypt/

It also fixes a form submit bug in IE.

### CODE REVIEW

**Comment by gbirke** [comment]
I've tested it now with IE8 - IE11 on Windows 7 and the crypto lib now works on all IE instances. Make sure to run `php combine_assets.php` before testing locally.

File: combine_assets.php [hunk 1]
```diff
@@ -4,12 +4,25 @@
...
+        'js/lib/function-bind.min.js',
         'js/banner/banner.js',
         'js/banner/banner.config.js',
         'js/banner/banner.encryption.js',
         'js/banner/banner.api.js',
         'js/banner/banner.form.js',
     ],
+    'js/lib/pgp-legacy-libs.js' => [
+        'js/lib/es6-promise.min.js',
+        'js/lib/base64.min.js', # The legacy PGP library is not compatible with the base64 polyfill so we have to include two libraries for base64 encoding :(
```
Comment (wmde-manicki) [inline combine_assets.php]: stupid question but is this polyfill actually used in pgp-legacy-libs? I remember adding it as `banner.api.js` was using `window.btoa`. And openpgpjs is surely using atob/btoa functions. But you're even saying in comment that atob provided by polyfill can't be used by the other encryption library

File: js/banner/banner.encryption.js [hunk 2]
```diff
@@ -7,11 +7,21 @@
 ( function ( Banner ) {
 	'use strict';
 
-	var EP;
+	var EP,
+		browserSupportsCryptoAPI = function () {
```
Comment (wmde-manicki) [inline js/banner/banner.encryption.js]: As long as I like this check in separate function, and I am not really sure what best Javascript conventions are, but why not declare this function as a "method" (ie. by declaring it as `EP.browserSupportsCryptoAPI = function()`? This seems to make sense to me as it'd be consistent with other parts of "Banner" library but if there are reasons not to make it then great

File: js/banner/banner.encryption.js [hunk 3]
```diff
@@ -7,11 +7,21 @@
...
+				&& typeof window.msCrypto.getRandomValues === 'function' ) {
+				return true;
+			}
+			return false;
+		};
 
 	function Encryption() {
 		var self = this;
 		this.initialized = false;
+		this.useLegacyEncryption = !browserSupportsCryptoAPI();
```
Comment (wmde-manicki) [inline js/banner/banner.encryption.js]: `useLegacyEncryption` is only used in `initCryptLib`, isn't it? If only used there, how about making it a local variable of that method?

**Comment by wmde-manicki** [comment]
So far I've tested it in Firefox (by forcing the use of the legacy library) and looked good.
I'd also like to have a look in IE once some slots are available on saucelabs
So +0.5 for now!

File: combine_assets.php (hunk 1 above)
Comment (gbirke) [inline combine_assets.php]: You're right, it is included for `banner.api.js` so it makes more sense to move it to the general lib.

**Comment by wmde-manicki** [comment]
Had a glimpse on IE8-IE11 on Wndows 7 (at Saucelabs), and encryption+validation seemed to work as expected. As mentioned in commit messages, IE8-IE10 use the legacy library, IE11 works with openpgpjs (maybe change the equality sign in the PR description for the future reference?).
Very nicely crafted @gbirke 

This is done from my point of view. @WMDE-Fisch @KaiNissen any comments/further thoughts?

**Pull request merged**